llmfoundry/data/finetuning/tasks.py

```diff
@@ -341,15 +345,23 @@ def __init__(self,
                 f'StreamingFinetuningDataset() got an unexpected keyword argument: {kwargs}'
             )
 
-        if remote is None or (local == remote):
-            if os.path.isdir(local):
-                contents = set(os.listdir(local))
-                if split not in contents:
-                    raise ValueError(
-                        f'local directory {local} does not contain split {split}'
-                    )
+        def _remote_local_validate(remote: Optional[str], local: Optional[str]):
```

Comment:
prefer defining this as a helper function not on the class since its totally independent